modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -10,6 +10,9 @@ import 'rxjs/add/operator/map';
 import * as models                                           from '../model/models';
 import { BASE_PATH, COLLECTION_FORMATS }                     from '../variables';
 import { Configuration }                                     from '../configuration';
+{{#withInterfaces}}
+import { I{{classname}} }                                    from './{{classname}}Interface';
```

Comment:
I agree with @Vrolijkx . The interface should have neither a suffix nor a prefix. The TS Guidelines only specify the 'I' prefix, but I think the idea behind it is, that prefixes & suffixes are unnecessary. 

I would propose to prefix the api implementation with `Default`.

For the `UserApi` the change would look like this:
class name: `DefaulUserApi`
Interface: `UserApi`